test/components/enketo-iframe.spec.js

```diff
@@ -99,4 +100,29 @@ describe('EnketoIframe', () => {
 
     postMessage.calledWith(data, window.location.origin).should.be.true;
   });
+
+  it('should not bubble up the message event if origin is different', async () => {
+    const wrapper = mountComponent({
+      props: { enketoId, actionType: 'new', instanceId: 'test-instance' },
+      container: {
+        router: mockRouter(`/?parentWindowOrigin=${window.location.origin}`)
+      }
+    });
+
+    sinon.stub($location, 'origin').value('https://example.com');
```

Comment:
I had the thought that instead of mocking the location, we could mock the entire event. Then again, what you're doing here is similar to what's done elsewhere in this file and is probably closer to how things would actually happen in production. It at least triggers the event from within the iframe. But if you wanted to avoid mocking the location, I think mocking the event is also an option, so I thought I'd mention it. It'd be something like:

```js
iframe.element.dispatchEvent(new MessageEvent('message', {
  origin: 'https://example.com',
  data
}));
```